tests/c/promote_many_funcs.c

```diff
@@ -0,0 +1,98 @@
+// Run-time:
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_PRINT_JITSTATE=1
+//   env-var: YKD_PRINT_IR=jit-post-opt
+//   stderr:
+//     jit-state: start-tracing
+//     y=201
+//     jit-state: stop-tracing
+//     --- Begin jit-post-opt ---
+//     ...
+//     define ptr @__yk_compiled_trace_0(...
+//       ...
+//       %{{condx1}} = icmp eq i64 {{jitx1}}, 111
+//       br i1 %{{condx1}}, label %{{succbbx1}}, label %{{failbbx1}}
+//       ...
+//       %{{condz1}} = icmp eq i64 {{jitz1}}, 89
+//       br i1 %{{condz1}}, label %{{succbbz1}}, label %{{failbbz1}}
+//       ...
+//       %{{add}} = add {{size_t}} %{{arg2}}, 89...
+//       ...
+//       %{{notconst2}} = load i64,...
+//       %{{y1}} = add {{size_t}} %{{add}}, %{{notconst2}}...
+//       ...
+//       %{{condx2}} = icmp eq i64 {{jitx2}}, 111
+//       br i1 %{{condx2}}, label %{{succbbx2}}, label %{{failbbx2}}
+//       ...
+//       %{{condz2}} = icmp eq i64 {{jitz2}}, 89
+//       br i1 %{{condz2}}, label %{{succbbz2}}, label %{{failbbz2}}
+//       ...
+//       %{{notconst4}} = udiv i64 %{{notconst3}}, 89...
+//       ...
+//       %{{y3}} = add {{size_t}} %{{y2}}, %{{notconst4}}...
+//       ...
+//     }
+//     ...
+//     --- End jit-post-opt ---
+//     y=402
+//     jit-state: enter-jit-code
+//     y=603
+//     y=804
+//     y=1005
+//     jit-state: deoptimise
+//     jit-state: exit-jit-code
+
+// Check that many-variable, many-function  promotion works in traces.
+//
+// FIXME: This test is showing that promotion is only on a best-effort basis.
```

Comment:
s/is only on a best-effort basis/is only a hint, rather than a guarantee/

Reply:
Fixed in Fixed in 96f196064a3dafea592a5e5a3da4c750e1a06e0a.